### PR TITLE
Adding tests

### LINKED ISSUE #1: Write some tests for ensuring correctness
We should write some test for proving that the app keeps doing what it is supposed to do.

I would recommend to write UI tests for decoupling from the implementation as we are going to try several approaches.

These tests should include:

* Empty case shows if no there are no todos.
* Todos should be rendered if the list is not empty.
* A Todo should be marked as Done (struck through) if clicked when it is not completed and viceversa.
* A Todo should be removed if clicked on the delete icon.
* A Todo should be added to the list after creating it. (This is a complex test as it involves navigation, form filling and navigating back).

### PR DESCRIPTION
**Pull request by Cotel**
Close #1 

I've added the 5 tests the issue details.

I haven't found the way to test `paintFlags` to see if a text view is stroken through so for that test I just perform the operations needed for arriving to the state I want and then I perform an assertion with the ServiceLocator which couples us a little bit to the implementation but I think we can use this for the moment.

I've also decided to make a recorded test for adding new todos as it involves many operations between screens and it was easier for me to record it.

### CODE REVIEW

File: app/src/androidTest/java/cotel/comonadic_ui/TodosListFragmentTest.kt [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+    todosListScreen {
+      todosList {
+        firstChild<TodosListScreen.TodoItem> {
+          click()
+
+          title {
+            hasText("One todo")
+
+            // Dunno how to test paint flags ¯\_(ツ)_/¯
```
Comment (tonilopezmr) [inline app/src/androidTest/java/cotel/comonadic_ui/TodosListFragmentTest.kt]: You need to create your own matcher, it is very easy.

```
public static Matcher<View> withTextColor(final int style) {
    Checks.checkNotNull(color);
    return new BoundedMatcher<View, TextView>(TextView.class) {
        @Override
        public boolean matchesSafely(TextView warning) {
            int currentStyle = getTypeface().getStyle();
            return style == currentStyle
        }
        @Override
        public void describeTo(Description description) {
            description.appendText("with style: ");
        }
    };
}
```

For example, I haven't tested but it should work something like that.

**Review by tonilopezmr — APPROVED** [review]
Thanks! It is easy! 👏👏👏👏👏

File: app/src/androidTest/java/cotel/comonadic_ui/TodosListFragmentTest.kt (hunk 1 above)
Comment (Cotel) [reply app/src/androidTest/java/cotel/comonadic_ui/TodosListFragmentTest.kt]: The thing here is that I'm using Kakao for defining a Screen pattern over the Fragment and make these assertions easier. I'll try to use an Espresso matcher inside Kakao. Thank you for the snippet 👍

Comment (tonilopezmr) [reply app/src/androidTest/java/cotel/comonadic_ui/TodosListFragmentTest.kt]: 👍 you are welcome

**Comment by tonilopezmr** [comment]
If you have added tests, It is a cool idea to add the `.travis.yml` file

File: app/src/androidTest/java/cotel/comonadic_ui/matchers.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import org.hamcrest.Matcher
+
+fun withPaintFlag(paintFlags: Int): Matcher<View> =
+  object : BoundedMatcher<View, TextView>(TextView::class.java) {
+    override fun describeTo(description: Description) {
+      description.appendText("with paint flag: ")
+    }
+
+    override fun matchesSafely(item: TextView): Boolean =
+      (item.paintFlags and paintFlags) == paintFlags
```
Comment (tonilopezmr) [inline app/src/androidTest/java/cotel/comonadic_ui/matchers.kt:18]: Cool!

**Comment by Cotel** [comment]
Dunno why travis is not triggering. Maybe it has to be in master at least once. I'm merging this and hope it triggers on the next PR.

**Pull request merged**